infor-design/enterprise-wc
# 1673 - IdsInputGroup input group validation

**Explain the details for making this change. What existing problem does the pull request solve?**
Created `IdsInputGroup` wrapper component to handle validating a group of input components with a shared group rule.

Similar to how custom validation is currently done by passing an `IdsValidationRule` into `IdsValidationMixin`,
IdsInputGroup accepts a custom validation with a simpler `IdsGroupValidationRule` as an argument into its
`setGroupValidationRule(rule)` api method. Check `example.ts` to see how it's used.

This PR also fixes the `change` event `IdsDropdown`, `IdsDatePicker`, `IdsTimePicker` which weren't firing.

**Related github/jira issue (required)**:
Closes #1673 

**Steps necessary to review your pull request (required)**:
1. Checkout branch
2. Go to http://localhost:4300/ids-input-group/example.html

**Test name example (first example)**
1. Type in the name `John Smith` or `Jane Doe` (middle initial not required)
2. A custom group warning message should appear
3. Type in any other name
4. There should be no warning message
5. Type in any name in lowercase
6. A custom error message should appear

**Test time picker example (second example)**
1. Enter value `9:00 AM` for Start Time
2. Enter value `7:00 PM` for End Time
3. A warning should appear if the range between start and end times is greater than 8 hours.
4. Enter value `4:55 PM` for End Time
5. There shouldn't be any warning message since time range is less than 8 hours.

**Test Styles**
1. Test against [figma design](https://www.figma.com/file/nbxdK0HViPxT1XPB2uUEUr/Validation-for-field-groups?type=design&node-id=1%3A621&mode=design&t=UM0GPCqA5b8Jq3dU-1)

**Test `change` event for components**
1. IdsDropdown
2. IdsTimePicker
6. IdsDatePicker

**Included in this Pull Request**:
